syntax="proto3";
package jarvis;


message jusers {
    string name     = 1;
    int32 right     = 2;
    string secret   = 3;
}

message time_table {
    int32  id      =   1;
    string date    =   2;
    double i       =   3;
    double s       =   4;
    double r       =   5;
    double b       =   6;
    double t       =   7;
}

message financial_users {
    int32 uid               = 1;
    string name             = 2;
    string create_time      = 3;
    string update_time      = 4;
}

message financial_records {
    int32 id            =  1;
    string when         =  2;
    int32 payer         =  3;
    int32 payee         =  4;
    double amount       =  5;
    int32  clasify      =  6;
    string comments     =  7;
    string create_time  =  8;
    string update_time  =  9;
    int32  status       = 10;  /* 软删除字段，0：有效，1：废弃 */
}

message financial_asset {
    int32 id            =   1;
    string timestamp    =   2;
    int32 uid           =   3;
    double amount       =   4;
    int32 recordid      =   5;
    string create_time  =   6;
    string update_time  =   7;
}

message tasks {
    int32 id         = 1;
    string date       = 2;
    string task       = 3;
    float esthour     = 4;
    float acthour     = 5;
    int32 priority   = 6;
    int32 completion = 7;
    int32 classify   = 8;
    int32 status     = 9;
    string comments   = 10;
    string ctime      = 11;
    string mtime      = 12;
}
